core/vtk/ttkArrayPreconditioning/ttkArrayPreconditioning.cpp

```diff
@@ -15,6 +15,9 @@
 vtkStandardNewMacro(ttkArrayPreconditioning);
 
 ttkArrayPreconditioning::ttkArrayPreconditioning() {
+#ifdef TTK_ENABLE_MPI
+  hasMPISupport_ = true;
```

Comment:
I'd recommend to move that to the constructor of the *base* class when available.
in this example `core/base/arrayPreconditioning/`